src/dialogs/config-flow/step-flow-abort.ts

```diff
@@ -33,6 +53,36 @@ class StepFlowAbort extends LitElement {
     `;
   }
 
+  private async _handleMissingCreds() {
+    if (
+      !(await showConfirmationDialog(this, {
```

Comment:
Why even open a new dialog and not just render this in the current step?

Reply:
It makes the code of data entry flow, which is used for auth/config flow/options flow tied to a very specific implementation for a marginally better user experience.

Reply:
I did initially implement this directly in config flow when testing it out, and it does work. By doing that, it is potentially easier for integrations to customize the steps (similar to the original nest config flow screen shots i've linked) using th existing translation mechanism.  We'd add a step into the config flow which renders the same thing as the app credentials dialog, then writes to the DB. It doesn't actually bother me that much that it's a different screen, but i realize it was different that the original arch discussion.  There has to be a hook/registration for app credentials platform into the config flow helper since I think the helper should not depend on the integration.

I think the frontend change is significantly less code to share the existing dialog, so I have come around to like this way better, but do acknowledge messing around with all these dialogs and making the frontend steps less generic in order to restart the config flow does add complexity.